x/committee/types/committee.go

```diff
@@ -13,52 +13,148 @@ import (
+	TallyOption      TallyOption   `json:"tally_option" yaml:"tally_option"`
+}
+
+// GetID is a getter for committee ID
+func (c BaseCommittee) GetID() uint64 { return c.ID }
+
+// GetDescription is a getter for committee Description
+func (c BaseCommittee) GetDescription() string { return c.Description }
+
+// GetPermissions is a getter for committee Permissions
+func (c BaseCommittee) GetPermissions() []Permission { return c.Permissions }
+
+// HasPermissionsFor returns whether the committee is authorized to enact a proposal.
+// As long as one permission allows the proposal then it goes through. Its the OR of all permissions.
+func (c BaseCommittee) HasPermissionsFor(ctx sdk.Context, appCdc *codec.Codec, pk ParamKeeper, proposal PubProposal) bool {
+	for _, p := range c.Permissions {
+		if p.Allows(ctx, appCdc, pk, proposal) {
 			return true
 		}
 	}
 	return false
 }
 
-// HasPermissionsFor returns whether the committee is authorized to enact a proposal.
-// As long as one permission allows the proposal then it goes through. Its the OR of all permissions.
-func (c Committee) HasPermissionsFor(ctx sdk.Context, appCdc *codec.Codec, pk ParamKeeper, proposal PubProposal) bool {
+// GetVoteThreshold is a getter for committee VoteThreshold
+func (c BaseCommittee) GetVoteThreshold() sdk.Dec { return c.VoteThreshold }
+
+// GetProposalDuration is a getter for committee ProposalDuration
+func (c BaseCommittee) GetProposalDuration() time.Duration { return c.ProposalDuration }
+
+// GetTallyOption is a getter for committee TallyOption
+func (c BaseCommittee) GetTallyOption() TallyOption { return c.TallyOption }
+
+// Validate validates BaseCommittee fields
+func (c BaseCommittee) Validate() error {
+	if len(c.Description) > MaxCommitteeDescriptionLength {
+		return fmt.Errorf("description length %d longer than max allowed %d", len(c.Description), MaxCommitteeDescriptionLength)
+	}
+
 	for _, p := range c.Permissions {
-		if p.Allows(ctx, appCdc, pk, proposal) {
+		if p == nil {
+			return fmt.Errorf("committee cannot have a nil permission")
+		}
+	}
+
+	if c.ProposalDuration < 0 {
+		return fmt.Errorf("invalid proposal duration: %s", c.ProposalDuration)
+	}
+
+	// threshold must be in the range (0,1]
+	if c.VoteThreshold.IsNil() || c.VoteThreshold.LTE(sdk.ZeroDec()) || c.VoteThreshold.GT(sdk.NewDec(1)) {
+		return fmt.Errorf("invalid threshold: %s", c.VoteThreshold)
+	}
+
+	return nil
+}
+
+func (c BaseCommittee) String() string {
+	return fmt.Sprintf(`Committee %d:
+  Description:              %s
+  Permissions:               			%s
+  VoteThreshold:            		  %s
+  ProposalDuration:        						%s
+  TallyOption:   						%d`,
+		c.ID, c.Description, c.Permissions,
+		c.VoteThreshold.String(), c.ProposalDuration.String(),
+		c.TallyOption,
+	)
+}
+
+//
+type MemberCommittee struct {
+	BaseCommittee `json:"base_committee" yaml:"base_committee"`
+	Members       []sdk.AccAddress `json:"members" yaml:"members"`
```

Comment:
`Members` should be added to the base type, as this defines who is allowed to make proposals. Given that, I think `MemberCommittee` can just be an alias for the base committee type.